doc/1.4/language.md

```diff
@@ -1642,8 +1642,16 @@ In the absence of explicit initializer expression, a default
 Note that the number of elements in the initializer must match with
 the number of elements or fields of the type of the *session*
 variable. This also implies that each sub-element, if itself being a
-compound data structure, must also be enclosed in braces. C99-style
-designated initializers are not supported.
+compound data structure, must also be enclosed in braces.
+
+C99-style designated initializers are supported for `struct`, `layout`, and
```

Comment:
Release note also mentions `extern struct`, but the difference is not significant to an end-user so makes sense to leave it out.

However, we *should* perhaps mention somewhere how struct initializers work for `extern` struct types, in particular when some fields are unknown to DML. My understanding is that such fields are initialized to 0.

Reply:
Correct. ISO standard specifies that when designated initializers are used, fields not initialized are initialized according to the rules for static variables, i.e. they become 0.

Reply:
I can add a blurb explaining this to the section in the DML manual which explains extern typedefs.